src/components/app/data/hooks/useAlgoliaSearch.test.tsx

```diff
@@ -0,0 +1,215 @@
+};
+const mockBaseAlgoliaData = {
+  securedAlgoliaApiKey: 'test-algolia-api-key',
+  catalogUuidsToCatalogQueryUuids: Object.fromEntries(mockCatalogUuidsToCatalogQueryUuids),
+};
+const mockBaseBFFData = {
+  ...mockBaseAlgoliaData,
+  enterpriseCustomer: mockEnterpriseCustomer,
+  errors: [],
+  warnings: [],
+};
+
+const mockBFFSearchData = {
+  ...mockBaseBFFData,
+};
+
+const mockBFFAcademyData = {
+  ...mockBaseBFFData,
+};
+
+const mockBFFSkillsQuizData = {
+  ...mockBaseBFFData,
+};
+
+const expectedOutputBasedOnIndex = ({ isMatchedRoute, isCatalogQueryFiltersEnabled, indexName }) => {
+  const unsupportedIndexes = [APP_CONFIG.ALGOLIA_INDEX_NAME_JOBS];
+  const isIndexSupported = !unsupportedIndexes.includes(indexName);
+  return isCatalogQueryFiltersEnabled && isIndexSupported && isMatchedRoute
+    ? {
+      catalogUuidsToCatalogQueryUuids: mockBaseAlgoliaData.catalogUuidsToCatalogQueryUuids,
+      searchClient: expect.objectContaining({
+        appId: APP_CONFIG.ALGOLIA_APP_ID,
+      }),
+      searchIndex: expect.objectContaining({
+        appId: APP_CONFIG.ALGOLIA_APP_ID,
+      }),
+    }
+    : {
+      catalogUuidsToCatalogQueryUuids: mockEmptyBaseAlgoliaData.catalogUuidsToCatalogQueryUuids,
+      searchClient: expect.objectContaining({
+        appId: APP_CONFIG.ALGOLIA_APP_ID,
+      }),
+      searchIndex: expect.objectContaining({
+        appId: APP_CONFIG.ALGOLIA_APP_ID,
+      }),
+    };
+};
+
+const matchQueryFnToURLPattern = ({ enterpriseSlug, queryFnToMatch }) => {
+  const queryFnToURLPattern = [{
+    route: `/${enterpriseSlug}`,
+    queryFn: fetchEnterpriseLearnerDashboard,
+  },
+  {
+    route: `/${enterpriseSlug}/search/`,
+    queryFn: fetchEnterpriseLearnerSearch,
+  },
+  {
+    route: `/${enterpriseSlug}/academies/academy-uuid`,
+    queryFn: fetchEnterpriseLearnerAcademy,
+  },
+  {
+    route: `/${enterpriseSlug}/skills-quiz`,
+    queryFn: fetchEnterpriseLearnerSkillsQuiz,
+  },
+  ];
+  const matchedQueryFnToRoute = queryFnToURLPattern.find(({ queryFn }) => queryFn === queryFnToMatch);
+  return matchedQueryFnToRoute?.route;
+};
+
+describe('useAlgoliaSearch', () => {
+  const Wrapper = ({
+    initialEntries = [], children,
+  }: { initialEntries?: string[] | undefined; children: React.ReactNode }) => (
+    <QueryClientProvider client={queryClient()}>
+      <MemoryRouter initialEntries={initialEntries}>
+        <Routes>
+          <Route path=":enterpriseSlug" element={children} />
+          <Route path=":enterpriseSlug/search" element={children} />
+          <Route path=":enterpriseSlug/academies/:academyUUID" element={children} />
+          <Route path=":enterpriseSlug/skills-quiz" element={children} />
+          <Route path=":enterpriseSlug/unsupported-bff-route" element={children} />
+        </Routes>
+      </MemoryRouter>
+    </QueryClientProvider>
+  );
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    // @ts-ignore
+    mockedUseEnterpriseCustomer.mockReturnValue({
```

Comment:
nit: I believe the following would get around the `// @ts-ignore`

```suggestion
    (mockedUseEnterpriseCustomer as jest.Mock).mockReturnValue({
```